android/src/main/kotlin/com/google/maps/flutter/navigation/GoogleMapView.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.maps.flutter.navigation
+
+import android.content.Context
+import android.view.View
+import com.google.android.gms.maps.GoogleMapOptions
+import com.google.android.gms.maps.MapView
+import io.flutter.plugin.platform.PlatformView
+
+class GoogleMapView
+internal constructor(
+  context: Context,
+  mapOptions: GoogleMapOptions,
+  viewId: Int,
+  viewEventApi: ViewEventApi,
+  private val viewRegistry: GoogleMapsViewRegistry,
+  imageRegistry: ImageRegistry
+) : PlatformView, GoogleMapsBaseMapView(viewId, viewEventApi, imageRegistry) {
+  private val _mapView: MapView = MapView(context, mapOptions)
```

Comment:
Should we use ViewStateAwareGMSMapView
here instead of MapView.

Related to comment:  https://github.com/googlemaps/flutter-navigation-sdk/pull/181#discussion_r1802686171